.gitlab-ci.yml

```diff
@@ -34,6 +35,63 @@ setup_spack_push:
             - spack 
         expire_in: 1 month
 
+#### CI builds of key4hep-stack
+# This is intended to run more complex ci builds, possibly comprising PRs in several repos
+# this job expects the following setup on the runner:
+# * environment variables GITHUB_TOKEN and GITHUB_USER
+#   set in .bashrc or via gitlab ci - to be able to query the commits on github
+#  
+build-spack-ci:
+    stage: compilation
+    tags:
+        - cvmfs
+    needs:
+        - "setup_spack_push"
+    parallel:
+        matrix:
+          - SPACKENV: 
+            - key4hep-ci
+    only:
+      refs:
+          - schedules # Only execute this on scheduled pipelines
+      variables:
+          - $K4_JOBTYPE == "CI"
+    script:
+        - |
+          #set up spack inside the k4-spack repo
+          source spack/share/spack/setup-env.sh
+          export KEY4HEP_RELEASE_VERSION=master-ci-`date -I`
+          python scripts/fetch_nightly_versions.py >> environments/$SPACKENV/spack.yaml
+          cd environments/$SPACKENV
+          source setup_clingo_centos7.sh
+          spack env activate .
+          spack repo add ../..
+          git clone https://gitlab.cern.ch/sft/sft-spack-repo.git
+          spack repo add ./sft-spack-repo
+          spack add key4hep-stack@${KEY4HEP_RELEASE_VERSION}
+          for K4_CI_URL in $K4_CI_URLS; do
+            # setup development repos
+            export PR_NUMBER=`echo $K4_CI_URL | grep -oP "pull/\K[0-9]*"`
+            export PR_REPO_URL=`echo $K4_CI_URL | grep -oP "\K.*/pull/"`
+            export PR_REPO_URL=`echo ${PR_REPO_URL/\/pull\//}`
+            export PACKAGE_NAME=`echo ${PR_REPO_URL} | rev | cut -d "/" -f -1 | rev `
+            export PACKAGE_NAME=`echo $PACKAGE_NAME |  tr '[:upper:]' '[:lower:]'`
+            echo "PACKAGE_NAME: ${PACKAGE_NAME}"
+            echo "PR_NUMBER: ${PR_NUMBER}"
+            echo "PR_REPO_URL: ${PR_REPO_URL}"
+            #sed -i "/${PACKAGE_NAME}@commit/d" ./spack.yaml
+            mkdir $PACKAGE_NAME; cd $PACKAGE_NAME
+            git clone $PR_REPO_URL .
+            #spack add ${PACKAGE_NAME}@commit.`git rev-parse HEAD`
+            spack develop -p $PWD ${PACKAGE_NAME}@commit.`git rev-parse HEAD`
+            git fetch origin pull/${PR_NUMBER}/head:testbranch${PR_NUMBER}
+            git checkout testbranch${PR_NUMBER}
```

Comment:
Yes, I think this will need some comments at least :) and I'm ok with putting this in a python script. Something like `spack pr-build`? Although it might be worthwhile discussing with upstream spack - I would suggest to just clearly comment the more arcane bash lines here for now and wait what happens there.